React/Views/RCTNavItem.m

```diff
@@ -11,75 +11,139 @@
+      if (_backButtonTitle && _backButtonTitle.length) {
+        _navigationItem.backBarButtonItem.title = _backButtonTitle;
+        return;
+      }
+      _navigationItem.backBarButtonItem = nil;
+    }
+  }
+  [self backButtonItem];
 }
 
 - (void)setBackButtonIcon:(UIImage *)backButtonIcon
 {
   _backButtonIcon = backButtonIcon;
-  _backButtonItem = nil;
+  if (_navigationItem.backBarButtonItem) {
+    if (_navigationItem.backBarButtonItem.image) {
+      if (_backButtonIcon) {
+        _navigationItem.backBarButtonItem.image = _backButtonIcon;
+        return;
+      }
+      _navigationItem.backBarButtonItem = nil;
+    }
+  }
+  [self backButtonItem];
 }
 
 - (UIBarButtonItem *)backButtonItem
 {
-  if (!_backButtonItem) {
-    if (_backButtonIcon) {
-      _backButtonItem = [[UIBarButtonItem alloc] initWithImage:_backButtonIcon
-                                                         style:UIBarButtonItemStylePlain
-                                                        target:nil
-                                                        action:nil];
-    } else if (_backButtonTitle.length) {
-      _backButtonItem = [[UIBarButtonItem alloc] initWithTitle:_backButtonTitle
-                                                         style:UIBarButtonItemStylePlain
-                                                        target:nil
-                                                        action:nil];
+  if (!_navigationItem.backBarButtonItem) {
+    if (_navigationItem.backBarButtonItem) {
+      _navigationItem.backBarButtonItem = [[UIBarButtonItem alloc] initWithImage:self.backButtonIcon
+                                                                           style:UIBarButtonItemStylePlain
+                                                                          target:nil
+                                                                          action:nil];
+    } else if (self.backButtonTitle.length) {
+      _navigationItem.backBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:self.backButtonTitle
+                                                                           style:UIBarButtonItemStylePlain
+                                                                          target:nil
+                                                                          action:nil];
     } else {
-      _backButtonItem = nil;
+      _navigationItem.backBarButtonItem = nil;
     }
   }
-  return _backButtonItem;
+  return _navigationItem.backBarButtonItem;
 }
 
 - (void)setLeftButtonTitle:(NSString *)leftButtonTitle
 {
   _leftButtonTitle = leftButtonTitle;
-  _leftButtonItem = nil;
+  if (_navigationItem.leftBarButtonItem) {
+    if (_navigationItem.leftBarButtonItem.title) {
+      if (_leftButtonTitle && _leftButtonTitle.length) {
+        _navigationItem.leftBarButtonItem.title = _leftButtonTitle;
+        return;
+      }
+      _navigationItem.leftBarButtonItem = nil;
+    }
+  }
+  [self leftButtonItem];
```

Comment:
Again, overcomplicated.

Also, there's a *lot* of copy-paste code here, with the same logic repeated for all three navigation buttons. Can't this be abstracted into a common function?